File: scripts/update_build.py
#!/usr/bin/env python3
"""Build version and date updater for index.html"""

import re
from datetime import datetime, timezone
import os

def update_build_info():
    """Update build version and date in index.html"""
    
    file_path = "index.html"
    
    if not os.path.exists(file_path):
        print(f"Error: {file_path} not found")
        return False
    
    try:
        # Read the file
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # Find and increment version
        version_match = re.search(r'meta name="build-version" content="([0-9.]+)"', content)
        if version_match:
            current_version = version_match.group(1)
            parts = current_version.split('.')
            parts[-1] = str(int(parts[-1]) + 1)
            new_version = '.'.join(parts)
            
            # Update meta tag
            content = re.sub(
                r'meta name="build-version" content="[^"]*"',
                f'meta name="build-version" content="{new_version}"',
                content
            )
            
            # Update footer version
            content = re.sub(
                r'<span id="version-value">V[0-9.]+</span>',
                f'<span id="version-value">V{new_version}</span>',
                content
            )
            
            print(f"✓ Version: {current_version} → {new_version}")
        else:
            print("✗ Error: Could not find build-version tag")
            return False
        
        # Update date to UTC
        utc_now = datetime.now(timezone.utc)
        utc_timestamp = utc_now.strftime("%Y-%m-%dT%H:%M:%SZ")
        
        # Format date for footer (e.g., "December 13, 2025")
        footer_date = utc_now.strftime("%B %d, %Y")
        
        # Update meta tag
        content = re.sub(
            r'meta name="build-date" content="[^"]*"',
            f'meta name="build-date" content="{utc_timestamp}"',
            content
        )
        
        # Update footer date
        content = re.sub(
            r'<span id="last-updated-value">[^<]*</span>',
            f'<span id="last-updated-value">{footer_date}</span>',
            content
        )
        
        print(f"✓ Date: {utc_timestamp}")
        
        # Write back
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(content)
        
        print("✓ index.html updated successfully!")
        return True
        
    except Exception as e:
        print(f"✗ Error: {e}")
        return False

if __name__ == "__main__":
    success = update_build_info()
    exit(0 if success else 1)

File: scripts/update_build.js
#!/usr/bin/env node

const fs = require('fs');

const FILE_PATH = 'index.html';

function formatDateUtc(date) {
  return date.toISOString().replace(/\.\d{3}Z$/, 'Z');
}

function formatDateFooter(date) {
  const month = new Intl.DateTimeFormat('en-US', { month: 'long', timeZone: 'UTC' }).format(date);
  const day = new Intl.DateTimeFormat('en-US', { day: '2-digit', timeZone: 'UTC' }).format(date);
  const year = new Intl.DateTimeFormat('en-US', { year: 'numeric', timeZone: 'UTC' }).format(date);
  return `${month} ${day}, ${year}`;
}

function getNextVersion(currentVersion) {
  const parts = currentVersion.split('.');
  const lastPart = Number.parseInt(parts[parts.length - 1], 10);
  if (Number.isNaN(lastPart)) {
    throw new Error('Current version is not numeric');
  }
  parts[parts.length - 1] = String(lastPart + 1);
  return parts.join('.');
}

function updateBuildInfo() {
  if (!fs.existsSync(FILE_PATH)) {
    console.error(`Error: ${FILE_PATH} not found`);
    return false;
  }

  try {
    let content = fs.readFileSync(FILE_PATH, 'utf8');

    const versionMatch = content.match(/meta name="build-version" content="([0-9.]+)"/);
    if (!versionMatch) {
      console.error('Error: Could not find build-version tag');
      return false;
    }

    const currentVersion = versionMatch[1];
    const newVersion = getNextVersion(currentVersion);

    content = content.replace(
      /meta name="build-version" content="[^"]*"/,
      `meta name="build-version" content="${newVersion}"`
    );
    content = content.replace(
      /<span id="version-value">V[0-9.]+<\/span>/,
      `<span id="version-value">V${newVersion}</span>`
    );

    console.log(`Version: ${currentVersion} -> ${newVersion}`);

    const nowUtc = new Date();
    const utcTimestamp = formatDateUtc(nowUtc);
    const footerDate = formatDateFooter(nowUtc);

    content = content.replace(
      /meta name="build-date" content="[^"]*"/,
      `meta name="build-date" content="${utcTimestamp}"`
    );
    content = content.replace(
      /<span id="last-updated-value">[^<]*<\/span>/,
      `<span id="last-updated-value">${footerDate}</span>`
    );

    console.log(`Date: ${utcTimestamp}`);

    fs.writeFileSync(FILE_PATH, content, 'utf8');
    console.log('index.html updated successfully!');
    return true;
  } catch (error) {
    console.error(`Error: ${error.message}`);
    return false;
  }
}

const success = updateBuildInfo();
process.exit(success ? 0 : 1);
